### ISSUE TITLE
MQTT User and Password Fields Limited to 30 characters each

### ISSUE THREAD

**Issue by withanhdammit** [Resinchem/ESP-Parking-Assistant#22]
I think the MQTT password field may be truncating my password.  I don't see anything in the code that would indicate that, and I know Arduino code can handle this length password (64 chars, letters & numbers only, no special chars).  I have another device early in my days of home automation where I built a temperature sensor that was pure MQTT, with the same password and it connects fine and still actively reports.

The Home Assistant MQTT Broker (Mosquitto) is reporting parkasst is not authorized.  I have double checked the password (it's copy/pasted so I know there's no typos.  It is a pretty long password, so maybe the textbox is truncating it?

```
2023-11-17 14:17:13: New connection from 10.42.100.113:52214 on port 1883.
2023-11-17 14:17:13: Client parkasst disconnected, not authorised.
```

**Comment by withanhdammit** [comment]
That's fantastic, thank you!

I've been looking through the code and haven't been able to spot where that's limited, so I appreciate you looking!

**Comment by withanhdammit** [comment]
I think I found the references (is the 31 for 30 chars plus CR?) I think this would do the trick:
Line 175
```char mqtt_pw[31];```
to
```char mqtt_pw[65];```

Line 765
```  char t_mqtt_pw[31];```
to
```  char t_mqtt_pw[65];```

Line 768
```  int user_pw = 31;```
to
```  int user_pw = 65;```

That should make the password field able to go up to 64 characters (plus CR (presumably))

**Comment by Resinchem** [comment]
Yes... those are the variable declarations, but without looking more closely at the code and testing, I'm not sure that just increasing those variable lengths will allow longer entries without causing other issues or errors.  There is also the web page entry sections and where it reads and writes to SPIFFS (e.g using mqttuser, and mqttpw.).  It would be great if it is that simple, but I would need to review all the code, make a test version and upload it to my test system to be sure it doesn't break something else.  I haven't worked on it in a while, so I just can't say right away if other changes may be needed or not.

If you have the ability to compile your own code, you are certainly free to try just changing those values and can let me know if it really is that simple and solves the issue.  Otherwise, I will try to get to it as soon as I get a break in my current project load.

**Comment by withanhdammit** [comment]
No worries, I'm happy to make those changes and test the new bin.  I'll let you know how it goes.  Hoping for no magic smoke LOL

**Comment by withanhdammit** [comment]
That did the trick!  I called my test version .44a, and I can see the MQTT messages now!
![image](https://github.com/Resinchem/ESP-Parking-Assistant/assets/16404870/479659ba-54c7-48a5-bbeb-8206d9435334)
![image](https://github.com/Resinchem/ESP-Parking-Assistant/assets/16404870/26151ad8-e955-436e-bfec-d8d24d847aea)

**Comment by Resinchem** [comment]
Outstanding!  Thanks for trying that and doing the hard work for me!  I'll leave this issue open until I roll those changes out in a new version, but you made the job much easier.

**Comment by Resinchem** [comment]
Do you happen to have a Discord account?  I'd like to reach out for a non-related issue.  If you have one and don't want to post it here (you can delete it afterwards if you want... I'll get a copy via email), you can always email me at the account you can find on the 'about' page of my YouTube channel.  If not, no big deal.  Just easier (and more appropriate) to chat there than here or in the comments of YouTube or the blog.

**Comment by withanhdammit** [comment]
I did some more work, specifically around getting these into a decent usable state in Home Assistant.

My HA/MQTT config is:
```
mqtt:
  sensor:
    - name: "Park Assist Distance"
      state_topic: "stat/parkmazda/parkdistance"
      unit_of_measurement: "in"
      icon: mdi:ruler
    - name: "Park Assist Car Detected"
      state_topic: "stat/parkmazda/cardetected"
    - name: "Park Assist MQTT Connected"
      state_topic: "stat/parkmazda/mqtt"

binary_sensor:
  - platform: template
    sensors:
      park_assist_car_detected:
        friendly_name: "Park Assist Car Detected"
        value_template: "{{ states('sensor.park_assist_car_detected') == '1' }}"
        icon_template: >-
          {% if is_state('binary_sensor.park_assist_car_detected', 'on') %}
            mdi:car
          {% else %}
            mdi:car-off
          {% endif %}
      park_assist_mqtt_connected:
        friendly_name: "Park Assist MQTT Connected"
        value_template: "{{ states('sensor.park_assist_mqtt_connected') == 'connected' }}"
        icon_template: >-
          {% if is_state('binary_sensor.park_assist_mqtt_connected', 'on') %}
            mdi:wifi
          {% else %}
            mdi:wifi-off
          {% endif %}
```
![image](https://github.com/Resinchem/ESP-Parking-Assistant/assets/16404870/f39541c0-900d-4ae2-9d56-f9809a87d366)


And my Lovelace cards look like this:
```
type: button
entity: binary_sensor.park_assist_car_detected
show_name: true
show_icon: true
tap_action:
  action: none
hold_action:
  action: none

type: button
entity: sensor.park_assist_distance
show_state: true
show_name: true
show_icon: true
tap_action:
  action: none
hold_action:
  action: none

type: button
entity: binary_sensor.park_assist_mqtt_connected
show_name: true
show_icon: true
tap_action:
  action: none
hold_action:
  action: none
```
![image](https://github.com/Resinchem/ESP-Parking-Assistant/assets/16404870/2f3f58bf-73fb-475f-b5c1-e883dbec6a42)
![image](https://github.com/Resinchem/ESP-Parking-Assistant/assets/16404870/3bdb4ce2-7ccc-4b0d-9716-48e23a30f8ef)


And to go really next level, it would be cool to have the thresholds available via MQTT, then I could code against those, too.  Oh, and the colors, so I could make the distance icon the same color as the thresholds, like if you're in the too far zone the icon is red, distance thresholds pulled from MQTT, if you're in the just right zone the color is green, and if you're in the too close zone the color is red.  Or something along those lines.  But future state :-D

**Comment by withanhdammit** [comment]
> Do you happen to have a Discord account? I'd like to reach out for a non-related issue.

I am on discord as 'withanHdammit'

**Comment by Resinchem** [comment]
Fixed with release v0.45.  MQTT User Name and Password can now be up to 64 characters each.
